### DISCUSSION TITLE
Generating a Canvas Patient Link

### DISCUSSION THREAD

**Discussion by tamalhu** [canvas-medical/canvas-plugins#285] [Q&A]
Hi Canvas team,

I'm working on the Bridge extension (https://github.com/Medical-Software-Foundation/canvas/tree/main/extensions/bridge) and we'd like to be able to generate links to Canvas patients that we can render on the Bridge platform.

I had a couple questions about how to achieve this functionality:

1. Is there a way to get the Canvas instance identifier from the SDK or other means? It seems like we'll need to know the Canvas instance to construct the link, e.g. for the test instance "xpc-dev", I have [xpc-dev.canvasmedical.com/patient/CANVAS_PATIENT_ID](http://xpc-dev.canvasmedical.com/patient/CANVAS_PATIENT_ID).

2. Related to the above, do Canvas partners CNAME their instances to their own custom domains? If so, then it might make sense to store the domain that will be used to construct the links as a secret value we can read in the extension.

**Comment by kristenoneill** [comment]
@tamalhu 

1. The customer identifier is not currently available - but we understand the need and captured the request in #290. At the moment, the best way to accomplish this would be to set the identifier as a secret that the plugin could use. We recognize that this is not scalable, but may be a way to unblock you in the interim. 

2. We do not allow customers to CNAME their instances.
